docs/source/guides/metrics.rst

```diff
@@ -0,0 +1,168 @@
+=======
+Metrics
+=======
+
+Metrics are measurements of statistics about your service, which can provide information about the usage and performance of your bentos in production.
+
+BentoML allows users to define custom metrics with `Prometheus <https://prometheus.io/>`_ to easily enable monitoring for their Bentos.
+ 
+This article will dive into how to add custom metrics to monitor your BentoService and how you can incorporate custom metrics into a :ref:`concepts/runner:Custom Runner`.
+
+Make sure to have `Prometheus <https://prometheus.io/download/#prometheus>`_ installed and running before continuing.
+
+Tracking model latency with Prometheus
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We will build a custom histogram to track the latency of our :ref:`custom MNIST runner <concepts/runner:Custom Runner>` in this
+tutorial.
+
+.. note::
+
+   The source code for this custom runner is :github:`available on GitHub <bentoml/BentoML/tree/main/examples/custom_model_runner>`.
+
+Initialize our Histogram to track inference duration:
+
+.. code-block:: python
+
+   import bentoml
+
+   inference_duration = bentoml.metrics.Histogram(
+       name="inference_duration",
+       documentation="Duration of inference",
+       labelnames=["torch_version", "device_id"],
+       buckets=exponential_buckets(0.001, 1.5, 10.0),
+   )
+
+This creates a :meth:`bentoml.metrics.Histogram`, which is a metric type that tracks the distribution of events with given bucket. The
+:attr:`bentoml.metrics.Histogram.buckets` is a exponential bucket with a factor of 1.5 and starts at 0.001.
+
+Follow with creating our custom MNIST runnable:
+
+.. code-block:: python
+
+   mnist_model = bentoml.pytorch.get("mnist_cnn:latest").
+   _BuiltinRunnable = mnist_model.to_runnable()
+
+   class MNISTRunnable(_BuiltinRunnable):
+        def __init__(self):
+            super().__init__()
+            import torch
+
+            print("Running on device:", self.device_id)
+            self.torch_version = torch.__version__
+            print("Running on torch version:", self.torch_version)
+
+        @bentoml.Runnable.method(batchable=True, batch_dim=0)
+        def __call__(self, input_arr: np.ndarray) -> np.ndarray:
+            start = time.perf_counter()
+            output = super().__call__(input_arr)
+            inference_duration.labels(
+                  torch_version=self.torch_version, device_id=self.device_id
+            ).observe(time.perf_counter() - start)
+            return output.argmax(dim=1)
```

Comment:
I think this example might be a bit too complex to demonstrate the metrics example. Can we maybe use a simpler example like `examples/custom_runner/nltk_pretrained_model`? Using the `nltk` example, we can also demonstrate the use of a `COUNTER` metric in the api_server when the return results is positive.

Reply:
Done.